workers/unity/Assets/Gdk/Core/GameObjectRepresentation/GameObjectReference.cs

```diff
@@ -0,0 +1,9 @@
+using UnityEngine;
+
+namespace Improbable.Gdk.Core
+{
+    public class GameObjectReference : Component
```

Comment:
Unless this is meant to exactly replicate something Unity would normally do for you. In which case that is fine.

Reply:
But it is being introduced in this PR. If we need to store a reference to a gameobject can we store a handle in an IComponentData and have a system own the manged data? Removing it is then part of our existing cleanup that already happens.

Reply:
In fact we already have the handle. Also this should go in playground as that seems like the place that consumes it. If we need it to be in the core for the moment that is fine though. Although we shouldn't have things across those boundaries.

Reply:
The handle is in Playground already. The GameObjectReference component itself needs to be in Core because that part is what enables the Reader/Writer APIs.

Reply:
Discussed elsewhere - this is fine for now, but will have to be revisited.